### PR TITLE
Fix seemingly random image ordering

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: shared/simplestreams/products.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type Product struct {
 	ReleaseTitle    string                    `json:"release_title"`
 	Supported       bool                      `json:"supported,omitempty"`
 	SupportedEOL    string                    `json:"support_eol,omitempty"`
+	Variant         string                    `json:"variant,omitempty"`
```
Comment (tomponline) [inline shared/simplestreams/products.go:33]: @stgraber worth adding a comment to the fields that diverge from the 'standard' simplestreams format?

File: shared/simplestreams/sort.go [hunk 2]
```diff
@@ -19,53 +18,43 @@ func (a sortedImages) Swap(i, j int) {
 }
 
 func (a sortedImages) Less(i, j int) bool {
-	if a[i].Properties["type"] != a[j].Properties["type"] {
-		if a[i].Properties["type"] == "squashfs" {
-			return true
+	for _, prop := range []string{"os", "release", "variant", "serial", "architecture"} {
```
Comment (tomponline) [inline shared/simplestreams/sort.go:21]: @stgraber you mentioned earlier that order was important when comparing these fields, if this is correct, could you add a comment to that effect here please.

File: shared/simplestreams/sort.go [hunk 3]
```diff
@@ -19,53 +18,43 @@ func (a sortedImages) Swap(i, j int) {
...
-		return false
+		return a[i].Properties[prop] < a[j].Properties[prop]
 	}
 
-	if a[j].Properties["os"] == "" {
-		return true
+	if a[i].Properties["type"] != a[j].Properties["type"] {
+		iScore := 0
+		jScore := 0
+		for score, pref := range []string{"squashfs", "root.tar.xz", "disk-kvm.img", "uefi1.img", "disk1.img"} {
```
Comment (tomponline) [inline shared/simplestreams/sort.go:44]: @stgraber is order important here too?

**Review by tomponline — APPROVED** [review]

**Comment by stgraber** [comment]
Updated to address comments

**Comment by stgraber** [comment]
@tomponline should be good to go

**Pull request merged**